logsmanagement/functions.c

```diff
@@ -0,0 +1,1118 @@
+    buffer_sprintf(wb, "\n         \"sticky\":%s,", (sticky)?"true":"false");                                   \
+    buffer_sprintf(wb, "\n         \"summary\":\"%s\",", summary);                                              \
+    buffer_sprintf(wb, "\n         \"filter\":\"%s\"", (range)?"range":"multiselect");                          \
+    buffer_sprintf(wb, "\n      }");                                                                            \
+    fields_added++;                                                                                             \
+} while(0)
+
+int logsmanagement_function_execute_cb( BUFFER *dest_wb, int timeout, 
+                                        const char *function, void *collector_data, 
+                                        void (*callback)(BUFFER *wb, int code, void *callback_data), 
+                                        void *callback_data) {
+
+    UNUSED(collector_data);
+    UNUSED(callback);
+    UNUSED(callback_data);
+
+    struct rusage start, end;
+    getrusage(RUSAGE_THREAD, &start);
+
+    logs_query_params_t query_params = {0};
+    unsigned long req_quota = 0;
+
+    unsigned int fn_off = 0, cn_off = 0;
+
+    while(function){
+        char *value = strsep_skip_consecutive_separators((char **) &function, " ");
+        if (!value || !*value) continue;
+        else if(!strcmp(value, "help")){
+            buffer_sprintf(dest_wb, FUNCTION_LOGSMANAGEMENT_HELP_LONG);
+            dest_wb->content_type = CT_TEXT_PLAIN;
+            return HTTP_RESP_OK;
+        }
+        else if (!strcmp(value, "sources")){
+            buffer_sprintf( dest_wb, 
+                            "{\n"
+                            "   \"api version\": %s,\n" 
+                            "   \"log sources\": {\n",
+                            LOGS_QRY_VERSION);
+            const logs_qry_res_err_t *const res_err = fetch_log_sources(dest_wb);
+            buffer_sprintf( dest_wb, 
+                            "\n"
+                            "   },\n"
+                            "   \"error code\": %d,\n"
+                            "   \"error\": \"%s\"\n}",
+                            (int) res_err->err_code,
+                            res_err->err_str);
+
+            return res_err->http_code;
+        }
+
+        char *key = strsep_skip_consecutive_separators(&value, ":");
+        if(!key || !*key) continue;
+        if(!value || !*value) continue;
+
+        // Kludge to respect quotes with spaces in between
+        // Not proper fix
+        if(*value == '_'){
+            value++;
+            value[strlen(value)] = ' '; 
+            value = strsep_skip_consecutive_separators(&value, "_");
+            function = strrchr(value, 0);
+            function++;
+        }
+
+        if(!strcmp(key, LOGS_MANAG_FUNC_PARAM_AFTER)){
+            query_params.req_from_ts = strtoll(value, NULL, 10);
```

Comment:
here you can also use our `str2ll`.

Reply:
`logsmanagement_function_execute_cb()` will be replaced by `logsmanagement_function_facets()`, it's only still there for testing, so I will mark this one as resolved.